src/rtc/rtc_s1_single_job.py

```diff
@@ -71,26 +73,145 @@ def update_mosaic_boundaries(mosaic_geogrid_dict, geogrid):
+           Browse image height
+       browse_image_width : int
+           Browse image width
+       scratch_dir : str
+           Directory for temporary files
+       temp_files_list: list (optional)
+           Mutable list of temporary files. If provided,
+           paths to the temporary files generated will be
+           appended to this list.
+       logger : loggin.Logger
+           Logger
+    """
 
+    BROWSE_IMAGE_MAX_PERCENTILE = 97
 
+    logger.info(f'creating browse image: {browse_image_filename}')
+
+    n_images = len(burst_imagery_list)
+
+    if n_images == 1:
+        expected_pol_order = pol_list
+    elif n_images == 2 and 'HH' in pol_list:
+        expected_pol_order = ['HH', 'HV']
+    elif n_images == 2:
+        expected_pol_order = ['VV', 'VH']
+    else:
+        expected_pol_order = ['HH', 'HV', 'VV']
+
+    alpha_channel = None
+    band_list = [None] * n_images
+
+    for filename, pol in zip(burst_imagery_list, pol_list):
+        logger.info(f'    pol: {pol}')
+        gdal_ds = gdal.Open(filename, gdal.GA_ReadOnly)
+        image_width = gdal_ds.GetRasterBand(1).XSize
+        image_height = gdal_ds.GetRasterBand(1).YSize
+
+        if (browse_image_height is not None or
+                browse_image_width is not None):
+
+            del gdal_ds
+
+            if browse_image_width is None:
+                browse_image_width =  int(np.round(
+                    (browse_image_height * float(image_width) / image_height)))
+
+            if browse_image_height is None:
+                browse_image_height = int(np.round(
+                    (browse_image_width * float(image_height) / image_width)))
+
+            logger.info(f'        browse length: {browse_image_height}')
+            logger.info(f'        browse width: {browse_image_width}')
+
+            browse_temp_file = tempfile.NamedTemporaryFile(
+                dir=scratch_dir, suffix='.tif').name
+
+            if temp_files_list is not None:
+                temp_files_list.append(browse_temp_file)
+
+            resamp_algorithm = 'AVERAGE'
+
+            # Translate the existing geotiff to the .png format
+            gdal.Translate(browse_temp_file,
+                           filename,
+                           # outputSRS="+proj=longlat +ellps=WGS84",
+                           # format='PNG',
+                           height=browse_image_height,
+                           width=browse_image_width,
+                           resampleAlg=resamp_algorithm)
+
+            gdal_ds = gdal.Open(browse_temp_file, gdal.GA_ReadOnly)
+
+        gdal_band = gdal_ds.GetRasterBand(1)
+        band_image = np.asarray(gdal_band.ReadAsArray(), dtype=np.float32)
+        if alpha_channel is None:
+            alpha_channel = np.asarray(np.isfinite(band_image),
+                                       dtype=np.float32)
+        vmax = np.nanpercentile(band_image, BROWSE_IMAGE_MAX_PERCENTILE)
+        logger.info('        min: 0')
```

Comment:
Is there any need to decide the lower limit of the image intensity using `numpy.nanpercentile` like it does for the upper limit?

Reply:
I usually use `0` for the lower limit (not a percentile/quantile).